dashboard/lib/policies/lti.rb

```diff
@@ -12,4 +15,39 @@ module AccessTokenScopes
   JWT_CLIENT_ASSERTION_TYPE = 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'.freeze
   JWT_ISSUER = CDO.studio_url('', CDO.default_scheme).freeze
   MEMBERSHIP_CONTAINER_CONTENT_TYPE = 'application/vnd.ims.lti-nrps.v2.membershipcontainer+json'.freeze
+  TEACHER_ROLES = Set.new(
+    [
+      'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Administrator',
+      'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Instructor',
+      'http://purl.imsglobal.org/vocab/lis/v2/membership#Administrator',
+      'http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor',
+      'http://purl.imsglobal.org/vocab/lis/v2/system/person#Administrator',
+    ]
+).freeze
+  LTI_ROLES_KEY = 'https://purl.imsglobal.org/spec/lti/claim/roles'.freeze
+  LTI_CUSTOM_CLAIMS = "https://purl.imsglobal.org/spec/lti/claim/custom".freeze
+
+  def self.get_account_type(id_token)
+    id_token[LTI_ROLES_KEY].each do |role|
+      return User::TYPE_TEACHER if TEACHER_ROLES.include? role
+    end
+    return User::TYPE_STUDENT
+  end
+
+  def self.generate_auth_id(id_token)
+    "#{id_token[:iss]}|#{id_token[:aud]}|#{id_token[:sub]}"
+  end
+
+  def self.lti?(user)
+    return !user.authentication_options.empty? && user.authentication_options.any?(&:lti?)
+    false
```

Comment:
This `false` is never reacher?

Reply:
Possibly the explicit return statement isn't needed either?

Reply:
Good catch